AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m

```diff
@@ -150,16 +132,16 @@ - (BOOL)startNotifier {
       if (SCNetworkReachabilityScheduleWithRunLoop(self.reachabilityRef,
                                                    CFRunLoopGetCurrent(),
                                                    kCFRunLoopDefaultMode)) {
-        returnValue = YES;
+        [[NSNotificationCenter defaultCenter] postNotificationName:kMSReachabilityChangedNotification
+                                                            object:self];
       }
     }
   };
   if ([NSThread isMainThread]) {
     block();
   } else {
-    dispatch_sync(dispatch_get_main_queue(), block);
+    dispatch_async(dispatch_get_main_queue(), block);
```

Comment:
I remember we make a *SYNC* call in main thread due to a deadlock issue. When `startNotifier` and `stopNotifier` calls come together it might cause the issue. Are you sure this is okay?

Reply:
Thinking about it, I don't like using the main thread I would assign reachability it's own serial queue and dispatch async the blocks there. That way we are sure not to bloat the main thread + everyone will use the same serial queue.

Reply:
@clpolet It was already implemented and then reverted.

@jaelim-ms **sync** cause deadlock because it LOCKS the current thread.

> When `startNotifier` and `stopNotifier` calls come together it might cause the issue.

Nope. It's queued in the main thread.